File: server/src/main/java/pt/iade/ei/thinktoilet/models/dtos/RatingCategory.java
package pt.iade.ei.thinktoilet.models.dtos;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class RatingCategory {
    private int id;
    private double clean;
    private double paper;
    private double structure;
    private double accessibility;

    public RatingCategory(int id, Double clean, Double paper, Double structure, Double accessibility) {
        this.id = id;
        this.clean = (clean != null) ? clean : 0.0f;
        this.paper = (paper != null) ? paper : 0.0f;
        this.structure = (structure != null) ? structure : 0.0f;
        this.accessibility = (accessibility != null) ? accessibility : 0.0f;
    }

    public RatingCategoryDTO toDTO() {
        return new RatingCategoryDTO(clean, paper, structure, accessibility);
    }
}

File: app/app/src/main/java/pt/iade/ei/thinktoilet/models/RatingCategory.kt
package pt.iade.ei.thinktoilet.models

import java.io.Serializable

data class RatingCategory(
    var clean: Float,
    var paper: Float,
    var structure: Float,
    var accessibility: Float
) : Serializable {
    fun average(): Float {
        return ((clean * 0.2f) + ((paper / 20) * 0.4f) + (structure * 0.2f) + accessibility * 0.2f)
    }
}
/*
Basicamente em fez de fazer uma média fizemos uma soma entre as variaveis e atribuimos um peso a cada uma delas
clean = 20%
paper = 40%
    dividimos por 20 para obter um valor entre 0 e 5 e depois multiplicamos por 40%
structure = 20%
accessibility = 20%
 */